tests/datasource/fluent/conftest.py

```diff
@@ -114,6 +152,176 @@ def inject_engine_lookup_double(
+    return result
+
+
+def _put_db_datasources_callback(
+    request: PreparedRequest,
+) -> _CallbackResult:
+    url = request.url
+    logger.info(f"{request.method} {url}")
+
+    item = _CLOUD_API_FAKE_DB.get(url, MISSING)
+    if not request.body:
+        errors = ErrorPayload(
+            errors=[{"code": "mock 400", "detail": "missing body", "source": None}]
+        )
+        result = _CallbackResult(400, headers=_DEFAULT_HEADERS, body=json.dumps(errors))
+    elif item is not MISSING:
+        payload = json.loads(request.body)
+        _CLOUD_API_FAKE_DB[url] = payload
+        result = _CallbackResult(
+            200, headers=_DEFAULT_HEADERS, body=json.dumps(payload)
+        )
+    else:
+        errors = ErrorPayload(
+            errors=[{"code": "mock 404", "detail": None, "source": None}]
+        )
+        result = _CallbackResult(404, headers=_DEFAULT_HEADERS, body=json.dumps(errors))
+
+    logger.info(f"Response {result.status}")
+    return result
+
+
+@pytest.fixture
+def cloud_api_fake():
+    org_url_base = f"{GX_CLOUD_MOCK_BASE_URL}/organizations/{FAKE_ORG_ID}"
+    dc_config_url = f"{org_url_base}/data-context-configuration"
+    datasources_url = f"{org_url_base}/datasources"
+
+    assert not _CLOUD_API_FAKE_DB, "_CLOUD_API_FAKE_DB should be empty"
+    _CLOUD_API_FAKE_DB.update(
+        {
+            dc_config_url: {
+                "anonymous_usage_statistics": {
+                    "data_context_id": FAKE_DATA_CONTEXT_ID,
+                    "enabled": False,
+                },
+                "datasources": {},
+            },
+            datasources_url: MISSING,
+        }
+    )
+
+    logger.info("Mocking the GX Cloud API")
+
+    with responses.RequestsMock(assert_all_requests_are_fired=False) as resp_mocker:
+        resp_mocker.add_callback(responses.GET, dc_config_url, _get_fake_db_callback)
+        resp_mocker.add_callback(
+            responses.GET,
+            f"{datasources_url}/{FAKE_DATASOURCE_ID}",
+            _get_fake_db_callback,
+        )
+        resp_mocker.add_callback(
+            responses.DELETE,
+            f"{datasources_url}/{FAKE_DATASOURCE_ID}",
+            _delete_fake_db_datasources_callback,
+        )
+        resp_mocker.add_callback(
+            responses.PUT,
+            f"{datasources_url}/{FAKE_DATASOURCE_ID}",
+            _put_db_datasources_callback,
+        )
+        resp_mocker.add_callback(
+            responses.POST, datasources_url, _post_fake_db_datasources_callback
+        )
+
+        yield resp_mocker
+
+    logger.info(f"Ending state ->\n{pf(_CLOUD_API_FAKE_DB, depth=1)}")
+    _CLOUD_API_FAKE_DB.clear()
+
+
+@pytest.fixture
+def empty_cloud_context_fluent(cloud_api_fake) -> CloudDataContext:
+    context = gx.get_context(
+        cloud_access_token=DUMMY_JWT_TOKEN,
+        cloud_organization_id=FAKE_ORG_ID,
+        cloud_base_url=GX_CLOUD_MOCK_BASE_URL,
+        cloud_mode=True,
+    )
+    return context
```

Comment:
Can you check with @billdirks to make sure this doesn't overlap with his work? I know he was looking for a proper Cloud mock so maybe y'all can work together

Reply:
Maybe I'll be able to use this. Currently, I'm mocking our cloud store calls (such as get) which wrap calls to cloud so I'm mocking above the response dependency.